### ISSUE TITLE
Problem displaying additional fields in simplified view

### ISSUE THREAD

**Issue by valentindragan** [pluginsGLPI/fields#285]
Steps to reproduce the bug
GLPI version 9.2.3
Fields version 1.7.3

Install a fresh instance of glpi 9.2.3
Install the Fields 1.7.3 plugin
Add a new block, with fields - (in my example I used two text fields)
http://prntscr.com/l4pyu7
Add a new user 
Add authorization Super Admin on root entity
Add authorization Self Service in root entity
Set Super Admin as default profile
Login with newly created user

**First occurence**

- Click Create Ticket - the fields **DO NOT** appear
- Click Create Ticket a second time - the fields appear
(I could reproduce this only in the fresh glpi install)

**Second occurrence**
- Logout
- Login with newly created user
- Immediately Switch profile to Self Service
- Click Create Ticket - the fields **DO NOT** appear
(you can click as many times, the fields won't appear)
- Switch back to Super Admin and try First Occurrence
- Switch back to Self Service 
- Click on Create Ticket - the fields **WILL** appear
**This also occurred in glpi 9.3.1 with Fields 1.8.2**

As far as I know, this is the last 9.2.3 compatible fields version, so I think a 1.7.4 version would really be necessary for solving this **High Importance bug**.

**Later edit:**
- The fields are not displayed when clicking the All Tab, in edit ticket view. This is another bug.
Workaround:
I changed this, in plugins/fields/inc/field.class.php
 `$subtype = $_SESSION['glpi_tabs'][strtolower($item::getType())];`
with this
`$subtype = $item::getType() == 'Ticket' ? 'Ticket$main':$_SESSION['glpi_tabs'][strtolower($item::getType())];`

**Comment by valentindragan** [comment]
Workaround:
Modify code on line 516 on fields/inc/fiels.class.php
as folows
 if (!isset($_SESSION['glpi_tabs'][strtolower($item::getType())])) {          
            if($item::getType() != 'Ticket')
            return;        
      };

**Comment by Twentycents84** [comment]
@valentindragan Hello
It's not working... i do it and the same behavior. 
i have the same problem :(

**Comment by valentindragan** [comment]
> @valentindragan Hello
> It's not working... i do it and the same behavior.
> i have the same problem :(
@Twentycents84 
Can you post a screenshot of your  updated code?
Is your problem related to tickets additional fields or other glpi types?

**Comment by valentindragan** [comment]
For Fields 1.8.2 the code to be replaced starts on line 514
**Please beware that this is an workaround, not a fix!** - I do not know what are the reasons for which that condition exists.
http://prntscr.com/l59fy2
Replace this
```
      if (!isset($_SESSION['glpi_tabs'][strtolower($item::getType())])) {
         return;
      };
```
with this

```
if (!isset($_SESSION['glpi_tabs'][strtolower($item::getType())])) {
    if($item::getType() != 'Ticket')
    return;
};
```

**Comment by Twentycents84** [comment]
Hello I do that but i have the same error comportment

**Comment by valentindragan** [comment]
I cannot help you without further details.
My workaround was for the exact problem and the exact environment i described above.
I wouldn't know about your fields and configuration.

**Comment by Twentycents84** [comment]
Hello What do you need exactly ?

**Comment by Twentycents84** [comment]
I have this  :  PHP Notice: Undefined index: entities_id in /var/www/html/glpi/plugins/fields/inc/field.class.php at line 547

**Comment by valentindragan** [comment]
You should not see the notice if debug is disabled and it should not affect our end result here.

**Comment by juveboss** [comment]
Hello,

the problem is fixed on version 1.9? Because it's a problem that have been coming from several versions..

Glpi : 9.3.1   Fields : 1.8.2 => KO

Thanks in advance

**Comment by andreytavares** [comment]
I'm having the same problem in version 1.91 in GLPI 9.4.1.1.

**Comment by juveboss** [comment]
Hello @cedric-anne 

You have a solution for glpi version 9.3.1 or 9.4.1.1 ?

i can give my log.. 

Thanks

**Comment by juveboss** [comment]
Hello,

You have a idea ?

Thanks

**Comment by mecmav** [comment]
I've done that, using the glpi 9.4.2 and the plugin_fields version 1.9.2. All of the option (URL, dropdown, text simple and multi-line yes/not )have been displayed in simplified interface, and in the previous versions of the glpi (< 9.4) and plugin_fields (< 1.9.2) really don't appear and don't work .

**Comment by mecmav** [comment]
> I cannot help you without further details.
> My workaround was for the exact problem and the exact environment i described above.
> I wouldn't know about your fields and configuration.



> For Fields 1.8.2 the code to be replaced starts on line 514
> **Please beware that this is an workaround, not a fix!** - I do not know what are the reasons for which that condition exists.
> http://prntscr.com/l59fy2
> Replace this
> 
> ```
>       if (!isset($_SESSION['glpi_tabs'][strtolower($item::getType())])) {
>          return;
>       };
> ```
> 
> with this
> 
> ```
> if (!isset($_SESSION['glpi_tabs'][strtolower($item::getType())])) {
>     if($item::getType() != 'Ticket')
>     return;
> };
> ```

this don't works in the version that you say

**Comment by abriuschi** [comment]
Hello all! I have the same problem! I added one fields, and sometimes appear the fields and sometimes doesn´t appear. In the simplified interface and in the standard interface too. 
It's a problem because the field I added is the desire date to solve the tickets.

Could you help me?
Thanks to all!

**Comment by juveboss** [comment]
Hello..

Up ! 

soon in a release?

**Comment by stonebuzz** [comment]
Hi all, 

try this PR https://github.com/pluginsGLPI/fields/pull/352 with latest version of plugin

which fix your problem.

Best regards

**Comment by stonebuzz** [comment]
Fix confirmed by #327
